File: app/models/BuzzerManager.java
package models;

import java.util.ArrayList;
import java.util.List;

public class BuzzerManager {

	private List<Team> respondedTeams = new ArrayList<Team>();
	
	public void reset() {
		this.respondedTeams = new ArrayList<Team>();
	}
	
	public List<Team> getRespondedTeams() {
		return respondedTeams;
	}
	
	public synchronized int respond(Team team) {
		respondedTeams.add(team);
		return respondedTeams.size();
	}
}

File: app/models/BuzzerManager.kt
package models

import java.util.*

class BuzzerManager {

    private var respondedTeams: MutableList<Team> = ArrayList()

    fun reset() {
        this.respondedTeams = ArrayList<Team>()
    }

    fun getRespondedTeams(): List<Team> {
        return respondedTeams
    }

    @Synchronized fun respond(team: Team): Int {
        respondedTeams.add(team)
        val responseOrder = respondedTeams.size
        team.buzzed(responseOrder)
        return responseOrder
    }
}
